modules/application-server-commons/src/main/java/org/testcontainers/applicationserver/ApplicationServerContainer.java

```diff
@@ -0,0 +1,415 @@
+
+    // How to query an application
+    private Integer httpPort;
+
+    private String appContextRoot;
+
+    // How to query application for readiness
+    private Integer readinessPort;
+
+    private String readinessPath;
+
+    // How long to wait for readiness
+    @NonNull
+    private Duration httpWaitTimeout = Duration.ofSeconds(60);
+
+    // Expected path for Microprofile platforms to query for readiness
+    static final String MP_HEALTH_READINESS_PATH = "/health/ready";
+
+    // Lazy environment variables
+    private Map<String, Supplier<String>> lazyEnvVars = new HashMap<>();
+
+    //Constructors
+
+    public ApplicationServerContainer(@NonNull final Future<String> image) {
+        super(image);
+    }
+
+    public ApplicationServerContainer(final DockerImageName dockerImageName) {
+        super(dockerImageName);
+    }
+
+    //Overrides
+
+    @Override
+    protected void configure() {
+        super.configure();
+
+        // Setup default wait strategy
+        waitingFor(
+            Wait.forHttp(readinessPath != null ? readinessPath : appContextRoot)
+                .forPort(readinessPort != null ? readinessPort : httpPort)
+                .withStartupTimeout(httpWaitTimeout)
+        );
+
+        // Copy applications
+        for(MountableFile archive : archives) {
+            withCopyFileToContainer(archive, getApplicationInstallDirectory() + extractApplicationName(archive));
+        }
+
+        // Add lazy env variables
+        for(Map.Entry<String, Supplier<String>> entry : lazyEnvVars.entrySet()) {
+            withEnv(entry.getKey(), entry.getValue().get());
+        }
+    }
+
+    @Override
+    protected void containerIsCreated(String containerId) {
+        if( Objects.isNull(tempDirectory) ) {
+            return;
+        }
+
+        try {
+            //Delete files in temp directory
+            for(String file : tempDirectory.toFile().list()) {
+                Files.deleteIfExists(Paths.get(tempDirectory.toString(), file));
+            }
+            //Delete temp directory
+            Files.deleteIfExists(tempDirectory);
+        } catch (IOException e) {
+            logger().info("Unable to delete temporary directory " + tempDirectory.toString(), e);
+        }
+
+    }
+
+    @Override
+    public void setExposedPorts(List<Integer> exposedPorts) {
+        if( Objects.isNull(this.httpPort) ) {
+            super.setExposedPorts(exposedPorts);
+            return;
+        }
+
+        super.setExposedPorts(appendPort(exposedPorts, this.httpPort));
+    }
+
+    //Configuration
+
+    /**
+     * One or more archives to be deployed to the application platform.
+     *
+     * Calling this method more than once will append new archives to a list to be deployed.
+     *
+     * @param archives - An archive created using shrinkwrap and test runtime
+     * @return self
+     */
+    public ApplicationServerContainer withArchvies(@NonNull Archive<?>... archives) {
```

Comment:
Fixed spelling error. 

I think having native ShrinkWrap support would really help with ease of use.  
If users are looking to replace Arquillian with TestContainers then those users will never have had to export their archives to the host system since Arquillian takes care of that behind the scenes.  

Without this kind of support I'd say it isn't even worth creating the ApplicationContainer and users could just use a GenericContainer.  ShrinkWrap is the value add here. 

> This way we can support those using ShrinkWrap and those who do not

This is achievable with the pre-existing method: 
```java
public ApplicationServerContainer withArchives(@NonNull MountableFile... archives)
```

Reply:
I built some examples for JakartaOne and ShrinkWrap play nice with Testcontainers by using `Transferable.of(byte[])`. 

https://github.com/eddumelendez/jakartaone-livestream-2023/blob/7bdccd7f7ac66bbb5903623f97dbff758b783c10/src/test/java/org/example/jakartaeelivestream2023/BaseTest.java#L13-L25

https://github.com/eddumelendez/jakartaone-livestream-2023/blob/7bdccd7f7ac66bbb5903623f97dbff758b783c10/src/test/java/org/example/jakartaeelivestream2023/PayaraMicroTest.java#L45-L46